### ISSUE TITLE
target: 'serverless' on local build

### ISSUE THREAD

**Issue by steida** [zeit/next.js#6144]
I just spend one hour investigating why next build does not work locally then I realized I have target set 'serverless' in next.config.js.

Maybe it should be ignored for the local build.

```
Error: Cannot find module '/Users/steida/dev/este/.next/server/static/jAvZuFRlVjcCJxcCa5aNH/pages/_document'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
```

**Comment by steida** [comment]
Or should we use this check?

`if (Boolean(process.env.NOW_REGION || process.env.NOW)) return` from https://github.com/zeit/now-env/search?q=process.env&unscoped_q=process.env

**Comment by timneutkens** [comment]
We should probably add an err.sh link when that error is thrown.

**Comment by chirag04** [comment]
@timneutkens thoughts on supporting non-minified and hot-reloading for local build in serverless target?

**Comment by marcinczenko** [comment]
Just to clarify: `next build` alone seems to be working just fine with `target: 'serverless'` in `next.config.js`. Only after running `next start` it logs `Error: Cannot find module '_document`' at the moment you try to access `localhost:3000` in the browser.

**Comment by timneutkens** [comment]
Sure, that's what I'm saying, we need to add an err.sh link to `next start` when that error is thrown, because it makes no sense to use `next start` after building a serverless build, that build has to be consumed by something other than Next.js.

**Comment by marcinczenko** [comment]
Do you already have an idea how would you like it to work so that we can have `serverless` deployment and still be able to run `next` for development?

Currently, I am just doing `yarn add next` when I develop, and then just before I deploy (or push) I do `yarn add next@canary` :). Fortunately `next@7` does not mind the `target: 'serverless'`.

**Comment by timneutkens** [comment]
You can use `next` for development just fine, `next start` is starting a *production server*, which doesn't fit serverless.

**Comment by steida** [comment]
As far I understand, ZEIT is working on `now dev`. My suggestions:

- run production server with serverless locally or allow easy configuration in next.config.js
- provide api to handle ports to /suffices mapping

Thank you!

**Comment by dav-is** [comment]
@marcinczenko this issue should be solved in `8.0.0-canary.17`

https://github.com/zeit/next.js/issues/6154

**Comment by marcinczenko** [comment]
Yes!!!! It works. Thanks!
